packages/web-forms/src/components/common/map/useMapBlock.ts

```diff
@@ -0,0 +1,293 @@
+		if (mapInstance) {
+			return;
+		}
+
+		mapInstance = new Map({
+			target: mapContainer,
+			layers: [new TileLayer({ source: new OSM() }), featuresVectorLayer],
+			view: new View({ center: DEFAULT_VIEW_CENTER, zoom: MIN_ZOOM }),
+			controls: [new Zoom()],
+		});
+
+		setCursorPointer();
+		currentState.value = STATES.READY;
+		loadGeometries(geoJSON);
+	};
+
+	const setCursorPointer = () => {
+		mapInstance?.on('pointermove', (event: MapBrowserEvent) => {
+			if (event.dragging || !mapInstance) {
+				return;
+			}
+
+			const hit = mapInstance.hasFeatureAtPixel(event.pixel, {
+				layerFilter: (layer) => layer instanceof WebGLVectorLayer,
+			});
+
+			mapInstance.getTargetElement().style.cursor = hit ? 'pointer' : '';
+		});
+	};
+
+	const handleClick = (event: MapBrowserEvent) => selectFeatureByPosition(event.pixel);
+
+	const toggleClickBinding = (bindClick: boolean) => {
+		mapInstance?.un('click', handleClick);
+
+		if (bindClick) {
+			mapInstance?.on('click', handleClick);
+		}
+	};
+
+	const fitToAllFeatures = (): void => {
+		const extent = featuresSource.getExtent();
+		if (extent) {
+			mapInstance?.getView().fit(extent, {
+				padding: [50, 50, 50, 50],
+				duration: ANIMATION_TIME,
+				maxZoom: MAX_ZOOM,
+			});
+		}
+	};
+
+	const centerCurrentLocation = (): void => {
+		// TODO: translations
+		const friendlyError = {
+			title: 'Cannot access location',
+			message:
+				'Grant location permission in the browser settings and make sure location is turned on.',
+		};
+
+		if (!navigator.geolocation) {
+			currentState.value = STATES.ERROR;
+			errorMessage.value = friendlyError;
+		}
+
+		navigator.geolocation.getCurrentPosition(
+			(position) => {
+				const coords = fromLonLat([position.coords.longitude, position.coords.latitude]);
+				mapInstance
+					?.getView()
+					.animate({ center: coords, zoom: MAX_ZOOM, duration: ANIMATION_TIME });
+				currentState.value = STATES.READY;
+			},
+			() => {
+				currentState.value = STATES.ERROR;
+				errorMessage.value = friendlyError;
+			},
+			{ enableHighAccuracy: true, timeout: GEOLOCATION_TIMEOUT_MS }
+		);
+	};
+
+	const centerFeatureLocation = (feature: Feature<GeometryType>): void => {
+		const geometry = feature.getGeometry();
+		const size = mapInstance?.getSize();
+		const view = mapInstance?.getView();
+
+		if (!geometry || !size?.length || !view) {
+			return;
+		}
+
+		const width = size[0];
+		const height = size[1];
+		let pixelOffsetY = 0;
+		let pixelOffsetX = -50;
+		if (width < SMALL_DEVICE_WIDTH) {
+			pixelOffsetY = 130;
+			pixelOffsetX = 0;
+		}
+
+		view.centerOn(geometry.getExtent(), size, [
+			width / 2 - pixelOffsetX,
+			height / 2 - pixelOffsetY,
+		]);
+	};
+
+	const loadGeometries = (geoJSON: FeatureCollection): void => {
+		if (!mapInstance) {
+			return;
+		}
+
+		currentState.value = STATES.LOADING;
+		selectFeature(undefined);
+		saveFeature(undefined);
+		featuresSource.clear();
```

Comment:
I've been calling `clear()` with argument `true`. The name of that parameter is `fast`, which sounds appealing haha. The API docs describe `fast` as:

> Skip dispatching of `removefeature` events.

As far as I know, those events aren't needed, so maybe `true` would help a little.